pkg/3scale/amp/component/apicast_hpa.go

```diff
@@ -0,0 +1,60 @@
+package component
+
+import (
+	"github.com/3scale/3scale-operator/pkg/helper"
+	hpa "k8s.io/api/autoscaling/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (apicast *Apicast) ApicastProductionHpa(namespace string, minPods *int32, maxPods int32, cpuPercent *int32, memoryPercent *int32) *hpa.HorizontalPodAutoscaler {
```

Comment:
IMO it would be better to have a single function for creating HPAs with addition of one extra field - Name.
No need to be replicating the function 3 times just to change the name IMO.

Reply:
Think I went down this road and was hitting some issues.  I can give it another crack to see what the issue was .

Reply:
had a go at this and the backend delete wasn't functioning correctly. Probally related to the points raised here https://github.com/3scale/3scale-operator/pull/901#discussion_r1423794896, I think over all a single hpa function makes sense and I will see if I can resolve the deletion issue.

Reply:
done